extension/src/project/readOnlyDocument.ts

```diff
@@ -87,19 +87,19 @@ export class ReadonlyDocument implements vscode.TextDocument {
         return ret;
     }
 
-    initialize(fileContent: string, languageId: string) {
-        this.fileContent = fileContent.split('\r\n').join('\n').split('\n').join('\r\n');//to make sure every line ends with \r\n
+    initialize(rawContent: string, langId: string) {
+        this.fileContent = rawContent.split('\r\n').join('\n').split('\n').join('\r\n');//to make sure every line ends with \r\n
```

Comment:
I've looked at this line several times before and always kind of wondered why we weren't doing something like this:
`rawContent.replace(/[^\r]\n/g, '\r\n')`

Reply:
Possibly because regular expressions are normally harder to maintain. I updated the code with your suggestion.